Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,552 @@
+            if (!bookmarkComponent->GetLocalBookmarksFileName().empty())
+            {
+                auto registry = AZ::SettingsRegistry::Get();
+                if (registry == nullptr)
+                {
+                    AZ_Warning(
+                        "ViewBookmarkLoader", false, "Unable to access global settings registry. Editor Preferences cannot be saved");
+                    return false;
+                }
+
+                auto projectUserRegistryPath = AZ::IO::FixedMaxPath(AZ::Utils::GetProjectPath()) / "user/Registry/ViewBookmarks";
+                projectUserRegistryPath /= AZ::IO::FixedMaxPathString(bookmarkComponent->GetLocalBookmarksFileName());
+
+                // Merge the current viewBookmark file into the settings registry.
+                bool isMerged = registry->MergeSettingsFile(
+                    projectUserRegistryPath.Native(), AZ::SettingsRegistryInterface::Format::JsonMergePatch, "/O3DE/ViewBookmarks");
+
+                if (isMerged)
+                {
+                    using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
+                    AZStd::string bookmarkKey = s_viewBookmarksRegistryPath + m_bookmarkfileName;
+                    auto viewBookmarkSettingsKey = FixedValueString::format(bookmarkKey.c_str());
+                    ViewBookmarkVisitor viewBookmarkVisitor;
+
+                    const bool visitedViewBookmarks = registry->Visit(viewBookmarkVisitor, viewBookmarkSettingsKey);
+
+                    if (visitedViewBookmarks)
+                    {
+                        m_localBookmarks = viewBookmarkVisitor.m_bookmarkMap.at(m_bookmarkfileName);
+                        m_localBookmarkCount = m_localBookmarks.size() - 1;
+                        m_lastKnownLocation = viewBookmarkVisitor.m_lastKnownLocation;
+                    }
+
+                    // once loaded we can remove the data from the settings registry.
+                    registry->Remove(bookmarkKey + "/");
+                    return visitedViewBookmarks;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    AZStd::optional<ViewBookmark> ViewBookmarkLoader::GetBookmarkAtIndex(int index, const StorageMode mode)
+    {
+        switch (mode)
+        {
+        case StorageMode::Shared:
+            AZ_Assert(false, "Shared Bookmark Component functionality not implemented.");
+            return AZStd::optional<ViewBookmark>();
+        case StorageMode::Local:
+
+            if (index >= 0 && index < m_localBookmarks.size())
+            {
+                return AZStd::optional<ViewBookmark>(m_localBookmarks.at(index));
+            }
+            AZ_Warning("ViewBookmarkLoader", false, "Couldn't load View Bookmark from file.");
+            return AZStd::optional<ViewBookmark>();
+        case StorageMode::Invalid:
+        default:
+            return AZStd::optional<ViewBookmark>();
+        }
+    }
```

Comment:
same case of unnecessary use of `optional<>` because the function always returns a valid optional.